### ISSUE TITLE
js: Add nats.Domain option

### ISSUE THREAD

**Issue by wallyqs** [nats-io/nats.go#746]
## Feature Request

Add option to set a JS domain for a jetstream context

```go
nats.Domain("HUB")
```

#### Use Case:

JS deployments with domains

#### Alternative Approaches

```go
js, err = nc.JetStream(nats.APIPrefix("$JS.HUB.API"))
```

**Comment by Jarema** [comment]
I created a #750  for this .

**Comment by wallyqs** [comment]
Fixed via #750
